pages/common/if.md

```diff
@@ -1,27 +1,28 @@
 # if
 
 > Simple shell conditional.
+> See also: `test`, `[`.
 
-- Echo a different thing depending on a command's success:
+- Executed two different commands based on a condition:
 
-`if {{command}}; then echo "success"; else echo "failure"; fi`
+`if {{condition}}; then {{echo "true"}}; else {{echo "false"}}; fi`
 
-- Full if syntax:
+- Check if a variable is defined:
 
-`if {{condition}}; then echo "true"; else echo "false"; fi`
+`if [[ ! -z {{$VARIABLE}} ]]; then {{echo "defined"}}; else {{echo "not defined"}}; fi`
```

Comment:
`! -z` won't test if a variable is defined, but more specifically if the the length of the variable value is non-zero. This is the same as using `-n`, which I suggest should be used instead.

Reply:
I didn't know about `-n`. 👍🏻   Yes, I know that it doesn't actually check if it's defined, but to a user, what's the difference between a variable that's not defined and a variable that has no value?
```suggestion
`if [[ -n {{$VARIABLE}} ]]; then {{echo "defined"}}; else {{echo "not defined"}}; fi`
```

Reply:
A variable can be defined or not, and have an empty value or not. These are two different things for a shell. This works in zsh:
```
% unset FOO
% if (( ${+FOO} )); then echo "FOO is defined"; else echo "FOO is not defined"; fi
FOO is not defined
% FOO=
% if (( ${+FOO} )); then echo "FOO is defined"; else echo "FOO is not defined"; fi
FOO is defined
% if [[ -n ${FOO} ]]; then echo "FOO is nonempty"; else echo "FOO is empty"; fi
FOO is empty
% FOO=1
% if [[ -n ${FOO} ]]; then echo "FOO is nonempty"; else echo "FOO is empty"; fi
FOO is nonempty
```
I know it's tricky...  :- )